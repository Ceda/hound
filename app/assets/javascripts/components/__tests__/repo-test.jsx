import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import Repo from '../repo.js';

it('renders a repo appropriately', () => {
  const repo = {
    id: 1,
    name: "Test repo",
    owner: {
      id: 1
    }
  }

  const onRepoClicked = jest.genMockFunction();

  const component = renderer.create(
    <Repo
      repo={repo}
      key={repo.id}
      onRepoClicked={onRepoClicked}
      isProcessingId={null}
    />

  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
