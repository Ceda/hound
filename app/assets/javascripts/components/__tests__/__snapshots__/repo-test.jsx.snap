exports[`test renders a repo appropriately 1`] = `
<li
  className="repo">
  <div
    className="repo-name">
    Test repo
  </div>
  <div
    className="repo-activation-toggle">
    <div
      className="repo-restricted">
      Only repo admins can activate
    </div>
  </div>
</li>
`;
